libyul/AssemblyStack.cpp

```diff
@@ -146,15 +147,17 @@ void AssemblyStack::optimize(Object& _object, bool _isCreation)
 	for (auto& subNode: _object.subObjects)
 		if (auto subObject = dynamic_cast<Object*>(subNode.get()))
 			optimize(*subObject, false);
-	EVMDialect const& dialect = dynamic_cast<EVMDialect const&>(languageToDialect(m_language, m_evmVersion));
-	GasMeter meter(dialect, _isCreation, m_optimiserSettings.expectedExecutionsPerDeployment);
-	OptimiserSuite::run(
-		dialect,
-		meter,
-		*_object.code,
-		*_object.analysisInfo,
-		m_optimiserSettings.optimizeStackAllocation
-	);
+	if (EVMDialect const* dialect = dynamic_cast<EVMDialect const*>(&languageToDialect(m_language, m_evmVersion)))
+	{
+		GasMeter meter(*dialect, _isCreation, m_optimiserSettings.expectedExecutionsPerDeployment);
+		OptimiserSuite::run(
+			*dialect,
+			meter,
+			*_object.code,
+			*_object.analysisInfo,
+			m_optimiserSettings.optimizeStackAllocation
+		);
+	}
```

Comment:
The intention here is to not run the optimizer if it's set to a non-evm dialect?

Reply:
Yes, not run the constant optimizer, because it only works on EVM for now.